network/internal/p2pfixtures/fixtures.go

```diff
@@ -135,6 +136,41 @@ func CreateNode(t *testing.T, networkKey crypto.PrivateKey, sporkID flow.Identif
 	return libp2pNode
 }
 
+// SubMustEventuallyStopReceivingAnyMessage checks that the subscription eventually stops receiving any messages within the given timeout by the context.
+// This func uses the send callback to continually publish messages to the subscription, this ensures that the subscription indeed stops receiving the messages.
+func SubMustEventuallyStopReceivingAnyMessage(t *testing.T, ctx context.Context, sub p2p.Subscription, publish func(t *testing.T)) {
+	done := make(chan struct{})
+	ticker := time.NewTicker(1 * time.Second)
+	defer func() {
+		close(done)
+		ticker.Stop()
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case <-ticker.C:
+				publish(t)
+			}
+		}
+	}()
+
+	// eventually we should stop receiving messages on the sub
+	require.Eventually(t, func() bool {
+		_, err := sub.Next(ctx)
+		return errors.Is(err, context.DeadlineExceeded)
+	}, 10*time.Second, 100*time.Millisecond)
+
+	// after we stop receiving messages on sub we should continue to not receiving messages
+	// despite messages continuing to be published
+	require.Never(t, func() bool {
+		_, err := sub.Next(ctx)
+		return !errors.Is(err, context.DeadlineExceeded)
+	}, 5*time.Second, 100*time.Millisecond)
+}
```

Comment:
Based on the comment `...we should continue to not receiving messages`, can we make a more strict assertion? e.g., 
```go
_, err := sub.Next(ctx)
require.ErrorIs(t, err, context.DeadlineExceeded)
```

Reply:
https://github.com/onflow/flow-go/pull/5355/commits/1691bb3698dc841ccd2abb15467df8bc1995ba5f